src/CachePlugin.php

```diff
@@ -379,7 +379,12 @@ private function createResponseFromCacheItem(CacheItemInterface $cacheItem)
 
         /** @var ResponseInterface $response */
         $response = $data['response'];
-        $response = $response->withBody($this->streamFactory->createStream($data['body']));
+        $stream = $this->streamFactory->createStream($data['body']);
+        if ($stream->isSeekable()) {
```

Comment:
:+1:  however we should still encapsulate it with a NetworkException to respect the contract